onmetal-apiserver/internal/apis/storage/validation/volume.go

```diff
@@ -118,6 +125,9 @@ func validateVolumeSpecUpdate(newSpec, oldSpec *storage.VolumeSpec, fldPath *fie
 
 	allErrs = append(allErrs, onmetalapivalidation.ValidateImmutableField(newSpec.VolumeClassRef, oldSpec.VolumeClassRef, fldPath.Child("volumeClassRef"))...)
 	allErrs = append(allErrs, onmetalapivalidation.ValidateSetOnceField(newSpec.VolumePoolRef, oldSpec.VolumePoolRef, fldPath.Child("volumePoolRef"))...)
+	if newSpec.Encryption != nil {
+		allErrs = append(allErrs, onmetalapivalidation.ValidateImmutableField(newSpec.Encryption.SecretRef, oldSpec.Encryption.SecretRef, fldPath.Child("encryption").Child("secretRef"))...)
+	}
```

Comment:
The entire `encryption` field should be immutable - it should not be possible to add encryption to a volume after it has been created.

Reply:
Agree. Done